cmd/tailscaled/tailscaled.service

```diff
@@ -20,5 +20,17 @@ CacheDirectory=tailscale
 CacheDirectoryMode=0750
 Type=notify
 
+LockPersonality=true
+MemoryDenyWriteExecute=true
+PrivateTmp=true
+ProtectControlGroups=true
+ProtectHome=true
```

Comment:
At some point if we implement filesharing things, we may need to remove this (or implement a multi-process architecture with different privilege layers), but for now this is fine.